### PR TITLE
Adding the ability to store microseconds in datetime string

### LINKED ISSUE #85: 🐛 [Datetime Format] Incorrect format for [Y-m-d H:i:s.u]
From https://github.com/spiral/database/issues/85

> https://github.com/spiral/database/blob/632488c5f7dfd3d4dc47edc9f1c2c992724e7526/src/Driver/Driver.php#L659
>
> I have the issue of inserting datetime in the following format `Y-m-d H:i:s.u `- `u` - dosent insert. 
>
> How can I insert my format to DB in another way?

### PR DESCRIPTION
**Pull request by msmakouz**

### CODE REVIEW

File: src/Config/MySQLDriverConfig.php [hunk 1]
```diff
@@ -32,6 +32,7 @@ public function __construct(
         bool $queryCache = true,
         bool $readonlySchema = false,
         bool $readonly = false,
+        bool $datetimeWithMicroseconds = false,
```
Comment (roxblnfk) [inline src/Config/MySQLDriverConfig.php]: WDYT about renaming it to `storeMicroseconds`?

I also think that option should be in ConnectionConfigs

File: src/Driver/Driver.php [hunk 2]
```diff
@@ -543,7 +544,19 @@ protected function formatDatetime(DateTimeInterface $value): string
...
+                (int) $value->format('n'),
+                (int) $value->format('j'),
+            )
+            ->setTime(
+                (int) $value->format('G'),
+                (int) $value->format('i'),
+                (int) $value->format('s'),
+                (int) $value->format('u'),
+            )
+            ->format($this->config->datetimeWithMicroseconds ? self::DATETIME_MICROSECONDS : self::DATETIME);
```
Comment (roxblnfk) [inline src/Driver/Driver.php]:
```suggestion
        try {
            $tz = $this->getTimezone();
        } catch (Throwable $e) {
            throw new DriverException($e->getMessage(), (int)$e->getCode(), $e);
        }
        return (clone $value)->setTimezone($tz)
            ->format($this->config->datetimeWithMicroseconds ? self::DATETIME_MICROSECONDS : self::DATETIME);
```

Can you test that code?

File: src/Driver/Driver.php [hunk 3]
```diff
@@ -539,24 +538,17 @@ protected function bindParameters(
     protected function formatDatetime(DateTimeInterface $value): string
     {
         try {
-            $datetime = new DateTimeImmutable('now', $this->getTimezone());
+            if ($value instanceof \DateTimeImmutable || $value instanceof \DateTime) {
+                $value = $value->setTimezone($this->getTimezone());
```
Comment (roxblnfk) [inline src/Driver/Driver.php]: In this case you will mutate `DateTime` instance.
Please also add a tests about it.

**Comment by msmakouz** [comment]
Closed PR from the fork. The code has been moved to a separate branch in the repository cycle/database.

**Comment by roxblnfk** [comment]
See https://github.com/cycle/database/pull/89

**Pull request closed without merging**